### PR TITLE
Grid built

### PR DESCRIPTION
**Pull request by Sergei-Nikolaienko**
https://Sergei-Nikolaienko.github.io/layout_catalog_grid/

### CODE REVIEW

**Comment by OlEzhA0** [comment]
hey! deploy your project
Screenshot 2022-12-02 at 1 28 47 PM

File: src/styles/_catalog.scss [hunk 1]
```diff
@@ -1,22 +1,36 @@
...
+}
 
-  &__card:not(:nth-child(4n)) {
-    margin-right: 48px;
+@media screen and (min-width: 1024px) {
+  .catalog {
+    display: grid;
+    grid-template-columns: repeat(4, $card-width);
   }
 }
```
Comment (OlEzhA0) [inline src/styles/_catalog.scss:36]: would be better to use scss `@each` here

File: src/styles/_catalog.scss [hunk 2]
```diff
@@ -1,22 +1,36 @@
...
+    height: $card-height;
     border: 1px solid #ccc;
     border-radius: 5px;
-    margin-bottom: 48px;
   }
+}
+
+@media screen and (min-width: 488px) {
+  .catalog {
+    display: grid;
```
Comment (OlEzhA0) [inline src/styles/_catalog.scss:19]: you shouldn’t define `display: grid` again

**Review by OlEzhA0 — CHANGES_REQUESTED** [review]
Hey! Good job! please check comments below

File: src/styles/_catalog.scss (hunk 1 above)
Comment (Sergei-Nikolaienko) [reply src/styles/_catalog.scss]: Hi, sorry, not sure how to implement @each here. I thought media queries have to be through the @media function

File: src/styles/_catalog.scss [hunk 3]
```diff
@@ -1,22 +1,33 @@
...
+  .catalog {
+    grid-template-columns: repeat(2, $card-width);
+  }
+}
+
+@media screen and (min-width: 768px) {
+  .catalog {
+    grid-template-columns: repeat(3, $card-width);
+  }
+}
```
Comment (AlisaVlasova) [inline src/styles/_catalog.scss:27]: move media into catalog block

**Review by AlisaVlasova — CHANGES_REQUESTED** [review]

File: src/styles/_catalog.scss [hunk 4]
```diff
@@ -1,22 +1,33 @@
...
+  @media screen and (min-width: 768px) {
+    .catalog {
+      grid-template-columns: repeat(3, $card-width);
+    }
+  }
+  
+  @media screen and (min-width: 1024px) {
+    .catalog {
+      grid-template-columns: repeat(4, $card-width);
+    }
```
Comment (OlEzhA0) [inline src/styles/_catalog.scss:22]: I suppose it will not work like this

File: src/styles/_catalog.scss [hunk 5]
```diff
@@ -1,22 +1,33 @@
...
+  gap: $grid-gap;
+  justify-content: center;
 
-  max-width: 944px;
-  margin: 0 auto;
+  @media screen and (min-width: 488px) {
+    .catalog {
+      grid-template-columns: repeat(2, $card-width);
+    }
+  }
```
Comment (OlEzhA0) [inline src/styles/_catalog.scss:11]:
```suggestion
  @media screen and (min-width: 488px) {
    grid-template-columns: repeat(2, $card-width);
  }
```

Comment (OlEzhA0) [reply src/styles/_catalog.scss:11]: and check it for each breakpoint

**Review by OlEzhA0 — CHANGES_REQUESTED** [review]
Hey! Check my comments below

**Comment by OlEzhA0** [comment]
Screenshot 2022-12-03 at 6 54 05 PM
I'm not sure it's our expected result

File: src/styles/_catalog.scss (hunk 4 above)
Comment (Sergei-Nikolaienko) [reply src/styles/_catalog.scss:22]: yup, my bad, fixing it asap

File: src/styles/main.scss [hunk 6]
```diff
@@ -10,5 +10,6 @@ body {
   box-sizing: border-box;
 }
 
+@import "./variables";
```
Comment (witflash) [inline src/styles/main.scss:13]: It's better to use the same syntax for all `@import`

File: src/styles/_catalog.scss [hunk 7]
```diff
@@ -1,22 +1,32 @@
...
 
-  max-width: 944px;
-  margin: 0 auto;
+  @media screen and (min-width: 488px) {
+    grid-template-columns: repeat(2, $card-width);
+
+  }
+
+  @media screen and (min-width: 768px) {
+
```
Comment (witflash) [inline src/styles/_catalog.scss:13]: There is no need to use empty line here

File: src/styles/_catalog.scss [hunk 8]
```diff
@@ -1,22 +1,32 @@
...
+    grid-template-columns: repeat(3, $card-width);
+
+  }
+
+  @media screen and (min-width: 1024px) {
+
```
Comment (witflash) [inline src/styles/_catalog.scss:19]: And here too

File: src/styles/_catalog.scss [hunk 9]
```diff
@@ -1,22 +1,32 @@
...
+  display: grid;
+  grid-template-columns: repeat(1, $card-width);
+  gap: $grid-gap;
+  justify-content: center;
 
-  max-width: 944px;
-  margin: 0 auto;
+  @media screen and (min-width: 488px) {
+    grid-template-columns: repeat(2, $card-width);
+
```
Comment (witflash) [inline src/styles/_catalog.scss:9]: Leftover (no unnecessary empty line)

**Review by witflash — APPROVED** [review]
Good job!

You can look at the last comments to improve your code in the future.

**Pull request closed without merging**